src/server/worker/pipeline/service/service.go

```diff
@@ -93,6 +96,30 @@ func forEachJob(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo, lo
 		if ji.State == pps.JobState_JOB_FINISHING {
 			return nil // don't pick up a "finishing" job
 		}
+		// Only restart the job once all source commits are closed.
```

Comment:
Should there be any checks for non-JOB_FINISHING states in the line above this one?

Reply:
no, we want to pick up all jobs that are active